source/localizable/tutorial/service.md

```diff
@@ -0,0 +1,302 @@
+  }
+});
+
+test('should append map element to container element', function(assert) {
+  this.set('myLocation', 'New York');
+  this.render(hbs`{{location-map location=myLocation}}`);
+  assert.equal(this.$('.map-container').children().length, 1, 'the map element should be put onscreen');
+  assert.equal(this.get('mapsService.calledWithLocation'), 'New York', 'a map of New York should be requested');
+});
+```
+
+In the `beforeEach` function that runs before each test, we use the implicit function `this.register` to register our stub service in place of the maps service.
+Registration makes an object available to your Ember application for things like loading components from templates and injecting services in this case.
+
+The call to the function `this.inject.service` injects the service we just registered into the context of the tests, so each test may access it through `this.mapsService`.
+In the example we assert that `calledWithLocation` in our stub is set to the location we passed to the component.
+
+To get the test to pass, add the container element to the component template.
+
+```app/templates/components/location-map.hbs
+<div class="map-container"></div>
+```
+
+Then update the component to append the map output to its inner container element.
+We'll add a maps service injection, and call the `getMapElement` function with the provided location.
+
+We then append the map element we get back from the service by implementing `didInsertElement`, 
+which is a [component lifecycle hook](../../components/the-component-lifecycle/#toc_integrating-with-third-party-libraries-with-code-didinsertelement-code).
+This function gets executed at render time after the component's markup gets inserted into the DOM.
+
+```app/components/location-map.js
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  maps: Ember.inject.service(),
+
+  didInsertElement() {
+    this._super(...arguments);
+    let location = this.get('location');
+    let mapElement = this.get('maps').getMapElement(location)
+    this.$('.map-container').append(mapElement);
+  }
+});
+```
+
+### Fetching Maps With a Service
+
+Now we have a passing component integration test, but no maps showing up when we view our web page.
+To generate the maps, we'll implement the maps service.
+
+Accessing our maps API through a service will give us several benefits
+
+* It is injected with a [service locator pattern](https://en.wikipedia.org/wiki/Service_locator_pattern), meaning it will abstract the maps API from the code that uses it, allowing for easier refactoring and maintenance
+* It is lazy-loaded, meaning it won't be initialized until it is called the first time.
+In some cases this can reduce your app's processor load and memory consumption.
+* It is a singleton, which will allow us cache map data.
+* It follows a lifecycle, meaning we have hooks to execute cleanup code when the app stopped, preventing things like memory leaks and unnecessary processing.
```

Comment:
when the app *stops*

Nicely done on explaining things here, I think that makes it far clearer to folks new to these ideas ...